components/constellation/constellation.rs

```diff
@@ -3851,6 +3865,41 @@ where
         }
     }
 
+    // Handle switching from fullscreen mode
+    fn switch_fullscreen_mode(&mut self, browsing_context_id: BrowsingContextId) {
+        if let Some(browsing_context) = self.browsing_contexts.get_mut(&browsing_context_id) {
+            let pipeline_id = browsing_context.pipeline_id;
+            let pipeline = match self.pipelines.get(&pipeline_id) {
+                None => {
+                    return warn!(
+                        "Pipeline {:?} switched from fullscreen mode after closing.",
+                        pipeline_id
+                    )
+                },
+                Some(pipeline) => pipeline,
+            };
+            let _ = pipeline
+                .event_loop
+                .send(ConstellationControlMsg::ExitFullScreen(pipeline.id));
+        }
+        // Send exit fullscreen message to any pending pipelines that aren't loaded yet
```

Comment:
This block isn't necessary. Only pipelines that have loaded will be in fullscreen mode.

Reply:
Sorry do you mean to delete if statement from switch_fullscreen_mode function?

Reply:
The loop over `self.pending_changes` is not needed.